test/parallel/test-child-process-chunked-stdout.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fork = require('child_process').fork;
+const stream = require('stream');
+const path = require('path');
+
+const producer = fork(path.join(common.fixturesDir, 'stdout-producer.js'),
+                      { silent: true });
+
+var found = '';
+const writable = new stream.Writable({
+  write(chunk, _, next) {
+    const lines = chunk.toString().split('\n');
+
+    found = lines[lines.length - 2];
+    next();
+  }
+});
```

Comment:
Wouldn't it suffice to collect the output to a string and then check that `output.endsWith('hey\n')`?